### PR TITLE
Issue901 multiple ext surf rect zone

### LINKED ISSUE #901: RectangularZoneTemplate does not allow multiple external surfaces connections per faces
I'll propose an implementation.

### PR DESCRIPTION
**Pull request by damienpicard**
This closes #901

### CODE REVIEW

**Comment by damienpicard** [comment]
The only strange thing is that when I try connect an exterior wall to proBusA, the graphical interface does not allow me to use index > 1. But if I do the connect from the editor, it works...

@Mathadon can you maybe have a look at that please? See added example.

**Comment by damienpicard** [comment]
@Mathadon there were still some bugs with indices + the non-reverse compatibility. I fixed them now, normally.

File: IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo [hunk 1]
```diff
@@ -1304,6 +1346,11 @@ components cannot be propagated.
...
 <ul>
 <li>
+August 28, 2018, by Damien Picard:<br/>
+Changes to allow multiple external surfaces connection per faces.
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/901\">#901</a>.
+</li>
+<li>
 August 26, 2018, by Damien Picard:<br/>
 Move all equations except windows equations of
 RectangularZoneTemplate to this interface model.
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo]: @damienpicard we discussed asserts that verify that the number of propsbusses is 1 when using an internal wall. Can you add these?

File: IDEAS/Resources/Scripts/Dymola/Buildings/Components/Examples/RectangularZoneExternalSurfaces.mos [hunk 2]
```diff
@@ -0,0 +1,4 @@
+simulateModel("IDEAS.Buildings.Components.Examples.RectangularZoneExternalSurfaces", startTime=0, stopTime=604800, method="Lsodar", tolerance=1e-06, resultFile="RectangularZoneExternalSurfaces");
```
Comment (Mathadon) [inline IDEAS/Resources/Scripts/Dymola/Buildings/Components/Examples/RectangularZoneExternalSurfaces.mos]: did you remove the old script?

Comment (damienpicard) [reply IDEAS/Resources/Scripts/Dymola/Buildings/Components/Examples/RectangularZoneExternalSurfaces.mos]: I renamed it. So normally it should work fine.

File: IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo (hunk 1 above)
Comment (damienpicard) [reply IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo]: Done

File: IDEAS/Resources/Scripts/Dymola/Buildings/Components/Examples/RectangularZoneExternalSurfaces.mos (hunk 2 above)
Comment (Mathadon) [reply IDEAS/Resources/Scripts/Dymola/Buildings/Components/Examples/RectangularZoneExternalSurfaces.mos]: ok!

Comment (Mathadon) [reply IDEAS/Resources/Scripts/Dymola/Buildings/Components/Examples/RectangularZoneExternalSurfaces.mos]: strange how unit tests pass despite that no reference results are included for these mos scripts..

File: IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo [hunk 3]
```diff
@@ -901,12 +919,24 @@ protected
   parameter Modelica.SIunits.Area ACei(fixed=false);
 
   final parameter Integer indWalA = if hasNoA then 0 else 1;
-  final parameter Integer indWalB = indWalA + (if hasNoB then 0 else 1);
-  final parameter Integer indWalC = indWalB + (if hasNoC then 0 else 1);
-  final parameter Integer indWalD = indWalC + (if hasNoD then 0 else 1);
-  final parameter Integer indFlo = indWalD + (if hasNoFlo then 0 else 1);
-  final parameter Integer indCei = indFlo + (if hasNoCei then 0 else 1);
-  final parameter Integer indIntZone_a = indCei + (if hasInt then 1 else 0);
+  final parameter Integer indWalA_end = indWalA + (if not hasNoA and bouTypA == IDEAS.Buildings.Components.Interfaces.BoundaryType.External
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo]: `bouTypA == IDEAS.Buildings.Components.Interfaces.BoundaryType.External` already implies that `hasNoA=true` so why did you add `not hasNoA and`?

File: IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo [hunk 4]
```diff
@@ -31,6 +31,25 @@ partial model RectangularZoneTemplateInterface
   parameter IDEAS.Buildings.Components.Interfaces.BoundaryType bouTypCei
     "Modelled boundary for the zone ceiling"
     annotation(Dialog(tab="Ceiling", group="Construction details"));
+  parameter Integer nExtA = 1 "Number of external surfaces connected to face A"
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo]: this could use `min=0`

**Comment by Mathadon** [comment]
@damienpicard this requires a merge :)

File: IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo (hunk 4 above)
Comment (damienpicard) [reply IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo]: done

File: IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo (hunk 3 above)
Comment (damienpicard) [reply IDEAS/Buildings/Components/Interfaces/RectangularZoneTemplateInterface.mo]: correct. Done
